src/Graphics/Texture.cs

```diff
@@ -30,6 +30,29 @@ public int LevelCount
 			protected set;
 		}
 
+		public override string Name
+		{
+			get
+			{
+				return base.Name;
+			}
+			set
+			{
+				// HACK: Avoid calling SetTextureName when the value hasn't changed.
+				if (value == base.Name)
+					return;
+
+				base.Name = value;
+
+				// HACK: Never pass a null string pointer through to SetTextureName.
+				// Since base.Name will be null by default, this will only happen if
+				//  you first set a name for the texture, then try to null it out.
+				if (value == null)
+					value = "";
```

Comment:
```suggestion
				// Avoid calling SetTextureName when the value hasn't changed.
				if (value == base.Name)
				{
					return;
				}

				base.Name = value;

				// Never pass a null string pointer through to SetTextureName.
				// Since base.Name will be null by default, this will only happen if
				//  you first set a name for the texture, then try to null it out.
				if (value == null)
				{
					value = string.Empty;
				}
```